src/entropy-keys.ts

```diff
@@ -1,17 +1,25 @@
+import type { Eip1024EncryptedData, Hex } from '@metamask/utils';
+import { bytesToHex, concatBytes } from '@metamask/utils';
+import { utf8ToBytes } from '@noble/ciphers/utils';
+import { x25519 } from '@noble/curves/ed25519';
 import { secp256k1 } from '@noble/curves/secp256k1';
 import { sha256 } from '@noble/hashes/sha256';
 
 import { addressToBytes, bytesToAddress } from './utils/address-conversion';
+import { ERC1024 } from './utils/ERC1024';
 
 /**
  * Retrieve the snap entropy private key.
+ * @param salt - Optional salt to use for the entropy derivation. Useful for generating keys for different purposes.
  * @returns Entropy Private Key Hex.
+ * @see https://metamask.github.io/SIPs/SIPS/sip-6
  */
-async function getEntropy(): Promise<`0x${string}`> {
+async function getEntropy(salt: string = ''): Promise<`0x${string}`> {
```

Comment:
Can we default the salt to undefined so we keep the same signing key?

```suggestion
async function getEntropy(salt?: string): Promise<`0x${string}`> {
```

Reply:
IIRC when we do not use a salt, the internal snap_getEntropy will use a default salt based on the snap that is invoking the RPC call.

(So other snaps can use getEntropy with no salt, and guarantees that snaps aren't all receiving the same private key)

Reply:
I updated the salt/entropy code to use the same logic related to origins as we use for signing.